File: GXpert/GXpert.Web/Modules/ServerTypes/Syllabus/CourseRow.ts
import { getLookup, getLookupAsync, fieldsProxy } from "@serenity-is/corelib";

export interface CourseRow {
    Id?: number;
    Title?: string;
    Description?: string;
    SortOrder?: number;
    Thumbnail?: string;
    IsActive?: boolean;
    InsertUserId?: number;
    InsertDate?: string;
    UpdateUserId?: number;
    UpdateDate?: string;
}

export abstract class CourseRow {
    static readonly idProperty = 'Id';
    static readonly nameProperty = 'Title';
    static readonly localTextPrefix = 'Syllabus.Course';
    static readonly lookupKey = 'Syllabus.Course';

    /** @deprecated use getLookupAsync instead */
    static getLookup() { return getLookup<CourseRow>('Syllabus.Course') }
    static async getLookupAsync() { return getLookupAsync<CourseRow>('Syllabus.Course') }

    static readonly deletePermission = 'Syllabus:SyllabusManagement:Modify';
    static readonly insertPermission = 'Syllabus:SyllabusManagement:Modify';
    static readonly readPermission = 'Syllabus:SyllabusManagement:View';
    static readonly updatePermission = 'Syllabus:SyllabusManagement:Modify';

    static readonly Fields = fieldsProxy<CourseRow>();
}

File: GXpert/GXpert.Web/Modules/Syllabus/Course/CourseRow.cs
using Serenity.ComponentModel;
using Serenity.Data;
using Serenity.Data.Mapping;
using Serenity.Extensions.Entities;
using System;
using System.ComponentModel;

namespace GXpert.Syllabus;

[ConnectionKey("Default"), Module("Syllabus"), TableName("Course")]
[DisplayName("Course"), InstanceName("Course")]
[ReadPermission(PermissionKeys.SyllabusManagement.View)]
[ModifyPermission(PermissionKeys.SyllabusManagement.Modify)]
[ServiceLookupPermission("Administration:General")]
[LookupScript("Syllabus.Course")]
public sealed class CourseRow : LoggingRow<CourseRow.RowFields>, IIdRow, INameRow
{
    [DisplayName("Id"), Identity, IdProperty]
    public int? Id { get => fields.Id[this]; set => fields.Id[this] = value; }

    [DisplayName("Title"), Size(500), NotNull, QuickSearch, NameProperty]
    public string Title { get => fields.Title[this]; set => fields.Title[this] = value; }

    [DisplayName("Description"), Size(2000)]
    public string Description { get => fields.Description[this]; set => fields.Description[this] = value; }

    [DisplayName("Sort Order")]
    public short? SortOrder { get => fields.SortOrder[this]; set => fields.SortOrder[this] = value; }

    [DisplayName("Thumbnail"), Size(500),FileUploadEditor]
    public string Thumbnail { get => fields.Thumbnail[this]; set => fields.Thumbnail[this] = value; }
    

    [DisplayName("Is Active"), NotNull,DefaultValue(1)]
    public bool? IsActive { get => fields.IsActive[this]; set => fields.IsActive[this] = value; }

    public class RowFields : LoggingRowFields
    {
        public Int32Field Id;
        public StringField Title;
        public StringField Description;
        public Int16Field SortOrder;
        public StringField Thumbnail;
        public BooleanField IsActive;

    }
}